tests/test_adjoint_jacobian.py

```diff
@@ -722,7 +785,7 @@ def circuit(x, y, z):
     @pytest.mark.parametrize("n_qubits", range(2, 6))
     @pytest.mark.parametrize("par", [-np.pi / 7, np.pi / 5, 2 * np.pi / 3])
     def test_controlled_jacobian(self, par, n_qubits, control_value, operation, tol):
-        """Test that the jacobian of the controlled gate matches the parameter-shift formula."""
+        """Test that the jacobian of the controlled gate matches the finite-diff formula."""
```

Comment:
Should we be using FD here or PS? Given the noise introduced by FD, I'd say PS may be a better long-term option, but will leave it to be your call (e.g. changing L817 or leaving as is)

Reply:
PS rules were missing for certain n-controlled gates, but they all seem to work fine now. I'll thus switch to PS. I'm keeping FD for `test_gate_jacobian` because PS fails for GlobalPhase.

Reply:
Wait, why does it fail for PS and not FD? Is that an issue to resolve?

Reply:
According to the [doc](https://docs.pennylane.ai/en/stable/code/api/pennylane.GlobalPhase.html), there is no gradient recipe, so it means PS will fail no?